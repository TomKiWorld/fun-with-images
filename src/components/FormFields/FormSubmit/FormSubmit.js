import React from 'react';

/**
 * Form submit
 * - Display short greet and amount of entries the user used the app
 * 
 * Required props:
 * - value => Displayed name
 * - type => Field type
 * - onClick => Function on button click
 * Optional props:
 * - extraClass => For classlist
 */
const FormSubmit = ({value, type, onClick, extraClass}) => {
  const receivedClass= extraClass ?? '';
  return (
    <div className='mt3'>
      <input 
        onClick={onClick}
        className={`${receivedClass} b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib`} 
        type={type}
        value={value}
      />
    </div>
  );
}

export default FormSubmit;
