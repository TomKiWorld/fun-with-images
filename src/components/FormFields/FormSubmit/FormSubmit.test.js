import { shallow } from 'enzyme';
import React from 'react';
import FormSubmit from './FormSubmit';

describe('FormSubmit Component', () => {
  it('Expect to render FormSubmit Component', () => {
    expect(shallow(<FormSubmit />)).toMatchSnapshot();
  });
});
