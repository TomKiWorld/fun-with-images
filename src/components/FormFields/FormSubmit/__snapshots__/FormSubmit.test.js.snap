// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSubmit Component Expect to render FormSubmit Component 1`] = `ShallowWrapper {}`;
